src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/Actions/UIInputFieldTextEntryAction.cs

```diff
@@ -0,0 +1,195 @@
+            {
+                return false;
+            }
+            
+            string normName = UIObjectPressAction.GetNormalizedGameObjectName(inputField.gameObject.name);
+            if (normName != NormalizedGameObjectName)
+            {
+                return false;
+            }
+
+            if (inputField is InputField legacyInputField)
+            {
+                if (!legacyInputField.isFocused)
+                {
+                    return false;
+                }
+            } else if (inputField is TMP_InputField tmpInputField)
+            {
+                if (!tmpInputField.isFocused)
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                throw new Exception("Unexpected object " + obj);
+            }
+
+            return true;
+        }
+
+        public override bool IsEquivalentTo(RGGameAction other)
+        {
+            if (base.IsEquivalentTo(other) && other is UIInputFieldTextEntryAction action)
+            {
+                return NormalizedGameObjectName == action.NormalizedGameObjectName;
+            }
+            return false;
+        }
+
+        public override IRGGameActionInstance GetInstance(Object obj)
+        {
+            return new UIInputFieldTextEntryInstance(this, obj);
+        }
+
+        protected override void WriteParametersToStringBuilder(StringBuilder stringBuilder)
+        {
+            stringBuilder.Append(",\n\"normalizedGameObjectName\":");
+            StringJsonConverter.WriteToStringBuilder(stringBuilder, NormalizedGameObjectName);
+        }
+
+        public override IEnumerable<(string, string)> GetDisplayActionAttributes()
+        {
+            foreach (var attr in base.GetDisplayActionAttributes())
+                yield return attr;
+
+            yield return ("Target Game Object", NormalizedGameObjectName);
+        }
+    }
+
+    public class UIInputFieldTextEntryInstance : RGGameActionInstance<UIInputFieldTextEntryAction, int>
+    {
+        public UIInputFieldTextEntryInstance(UIInputFieldTextEntryAction action, Object targetObject) : base(action, targetObject)
+        {
+        }
+
+        protected override bool IsValidActionParameter(int param)
+        {
+            return true;
+        }
+
+        private IEnumerable<Key> TextEntryKeys()
+        {
+            for (Key key = UIInputFieldTextEntryAction.MinKey; key <= UIInputFieldTextEntryAction.MaxKey; ++key)
+            {
+                yield return key;
+            }
+        }
+
+        protected override IEnumerable<RGActionInput> GetActionInputs(int param)
+        {
+            if (param == UIInputFieldTextEntryAction.PARAM_NULL)
+            {
+                foreach (Key key in TextEntryKeys())
+                {
+                    yield return new InputSystemKeyInput(key, false);
+                }
+                yield return new InputSystemKeyInput(Key.Backspace, false);
```

Comment:
Can you point to an example where this method is called.  It looks unusual with all the chained yield return statements and I'd like to better understand the full usage pattern

Reply:
This ultimately gets called by the bot to determine the set of keyboard inputs to perform on a particular frame. I've decided on a coroutine API for this because for some bots they don't necessarily need to store all the inputs in a buffer, so they could just iterate over all the inputs for that action, e.g.
```
foreach (RGActionInput input in chosenAction.GetInputs(actionParam))
  input.Perform();
```

In the default `RGMonkeyBot` implementation it stores all the inputs into a buffer for that frame:
https://github.com/Regression-Games/RGUnityBots/blob/be91bd43937b64e38da6bed2f9f58027ba0a1ee9/src/gg.regression.unity.bots/Runtime/Scripts/GenericBots/RGMonkeyBotLogic.cs#L174

Reply:
Ok, then lets add some big summary code docs around that interface explaining the usage pattern for extenders / implementers, probably including at least one sample code snippet

Reply:
I'm working on a technical reference here on another branch that will reside within the action manager directory for all this work, I think this should cover it? 
https://github.com/Regression-Games/RGUnityBots/blob/sasha/reg-1873/src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/README.md

Reply:
I've created a PR with the technical reference here https://github.com/Regression-Games/RGUnityBots/pull/238. Do let me know if you think we should still add an inline comment with this type of example elsewhere.

Reply:
Even if we have the docs, we should have the correct <summary> docs on our APIs and interfaces to enable users within their IDEs and when reading our source.

Reply:
I've added more usage examples and comments to the summary docs in the code for RGGameAction and related APIs.